source/backend/database/models/completedRecipesModel.js

```diff
@@ -13,4 +13,18 @@ async function getCompletedRecipes(userId){
   return completedRecipes;
 }
 
-module.exports = { getCompletedRecipes };
\ No newline at end of file
+/**
+ * Get all completed challenges 
+ * @param challenge
+ * @returns {Promise<awaited Knex.QueryBuilder<TRecord, TResult>>}
+ */
+async function getCompletedChallenges(userId, challenge) {
+  const completedChallenges = await db('completedRecipes')
+    .join('users', 'completedRecipes.userId', 'users.userId')
+    .join('recipes', 'completedRecipes.recipeId', 'recipes.recipeId')
+    .select('recipes.*')
+    .where({userId, challenge});
+  return completedChallenges; 
+}
+
+module.exports = { getCompletedRecipes, getCompletedChallenges};

```

Comment:
We talked about this earlier, but i think this makes sense. We'll have Thomas take a look at this to make sure, but I think it's good.